File: SEP3_ChatSystem/Model/Unit/Message/PrivateMessage.cs
namespace SEP3_ChatSystem.Model.Unit.Message
{
    public class PrivateMessage : Message
    {
        public string ReceiverId { get; set; }

        public PrivateMessage(string senderId, string receiverId, string messageInfo) : base(senderId, messageInfo,
            MessageType.PRIVATE)
        {
            ReceiverId = receiverId;
        }
        
        public PrivateMessage() {}

        public PrivateMessage Copy()
        {
            return new PrivateMessage
            {
                MessageInfo = MessageInfo,
                Type = Type,
                SenderId = SenderId,
                Time = Time,
                ReceiverId = ReceiverId
            };
        }

        public override bool CanRead(string userId)
        {
            return userId == SenderId || userId == ReceiverId;
        }
    }
}

File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/unit/message/PrivateMessage.java
package com.example.SEP3_Database.model.domain.unit.message;

import com.example.SEP3_Database.model.domain.unit.time.MyTime;

public class PrivateMessage extends Message{
    private String ReceiverId;

    public PrivateMessage(String senderId, String receiverId, String messageInfo)
    {
        super(senderId, messageInfo, MessageType.PRIVATE);
        ReceiverId = receiverId;
    }

    public PrivateMessage(String senderId,String receiverId,String messageInfo,MyTime time)
    {
        super(senderId, messageInfo, MessageType.PRIVATE,time);
        ReceiverId = receiverId;
    }

    public String getReceiverId() {
        return ReceiverId;
    }

    public void setReceiverId(String receiverId) {
        ReceiverId = receiverId;
    }

    public PrivateMessage copy()
    {
        return new PrivateMessage(super.SenderId,ReceiverId,super.MessageInfo,super.Time);
    }
}
